### ISSUE TITLE
Regression since 3.0.3 with exports

### ISSUE THREAD

**Issue by JSteunou** [lgarron/clipboard-polyfill#150]
I was updating from 3.0.1 but I have this error at build with webpack with 3.0.3

`[webpack-cli] ModuleNotFoundError: Module not found: Error: Package path ./src/implementations/write-fallback is not exported from package /myproject/node_modules/clipboard-polyfill (see exports field in /myproject/node_modules/clipboard-polyfill/package.json)`

it seems the new `exports` field solve some issues, but create others. I'm not an expert on this, sorry I have no fix to suggest.

**Comment by lgarron** [comment]
That sounds like `webpack` is trying to use the source code, which sounds pretty unexpected to me. I don't use `webpack`, so I would need a repro case to have a chance of tackling this. Do you have a repo/branch where this shows up?

**Comment by lgarron** [comment]
> I don't use `webpack`, so I would need a repro case to have a chance of tackling this. Do you have a repo/branch where this shows up?

This would still be very helpful!

**Comment by JSteunou** [comment]
Sorry no bandwith atm, will try asap.

**Comment by JounQin** [comment]
It's Node spec, `exports` makes unexported paths `private` and unavailable from other package.

**Comment by lgarron** [comment]
> It's Node spec, `exports` makes unexported paths `private` and unavailable from other package.

Sure, but relative imports inside the project should still work. And the three support paths are explicitly exported here:
https://github.com/lgarron/clipboard-polyfill/blob/06404f6ffc539720f8d83a5165c7169d45e7af2a/package.json#L67-L83

@JSteunou: What are you trying to import?

**Comment by JSteunou** [comment]
```js
import {writeText, readText} from 'clipboard-polyfill'
import {writeFallback} from 'clipboard-polyfill/src/implementations/write-fallback'

```

**Comment by lgarron** [comment]
@JSteunou That would explain it. You should not import `clipboard-polyfill/src/implementations/write-fallback`, it is not a public API and not meant to be possible to import. Are you able to replace your use of `writeFallback` with `writeText`?

**Comment by JSteunou** [comment]
Ho ok. Unfortunately we have to try catch `writeText` and fallback to `writeFallback` in case of failure. Failures happens in crossorigin iframe (an app in Teams for example).

**Comment by lgarron** [comment]
> Ho ok. Unfortunately we have to try catch `writeText` and fallback to `writeFallback` in case of failure. Failures happens in crossorigin iframe (an app in Microsoft Teams for example).

Hmm, sit sounds like what would be more useful in your case is for the library to try `writeFallback` if copying fails for any reason. I'd be hesitant to change that at this point, though.

Fro the time being, I would recommend pinning to an earlier version without the `exports` field.

**Comment by lgarron** [comment]
Could you let me know if this is fixed for you in `v4.0.0-rc3`?

**Comment by lgarron** [comment]
> Hmm, it sounds like what would be more useful in your case is for the library to try `writeFallback` if copying fails for any reason. I'd be hesitant to change that at this point, though.

This is now the case, so I'm tentatively closing this as resolved. Please feel welcome to file a new issue if you're still having problems!
